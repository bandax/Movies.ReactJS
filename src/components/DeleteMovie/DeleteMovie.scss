.delete-movie-window {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  transition: opacity 500ms;
}

.delete-movie-modal {
  margin: 70px auto;
  padding: 20px;
  background: #000000;
  border-radius: 5px;
  width: 30%;
  position: relative;
  transition: all 5s ease-in-out;
}

.delete-movie-title {
  margin: 10px 0 20px 0;
  color: #ffffff;
  text-transform: uppercase;
  font-size: 35px;
  font-weight: bold;
}

.delete-movie-close {
  position: absolute;
  top: 5px;
  right: 30px;
  transition: all 200ms;
  font-size: 30px;
  font-weight: bold;
  text-decoration: none;
  color: #ffffff;
  font-size: 40px;
}

.delete-movie-content {
  max-height: 30%;
  overflow: auto;
  color: #ffffff;
  font-size: 18px;
  margin: 0 0 10px 0;
}

.delete-movie-actions {
  text-align: right;
}

.btn-confirm {
  background-color: #f65261;
  color: white;
  font-size: 18px;
  padding: 10px 50px;
  border-radius: 5px;
  text-transform: uppercase;
}

@media screen and (max-width: 700px) {
  .box {
    width: 70%;
  }
  .delete-movie-modal {
    width: 70%;
  }
}
