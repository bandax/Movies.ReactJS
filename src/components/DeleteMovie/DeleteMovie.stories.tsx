import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import { DeleteMovieWindow, IDeleteMovieWindowProps } from './DeleteMovie';

export default {
  title: 'Movies/Delete Movie',
  component: DeleteMovieWindow,
} as Meta;

const Template: Story<IDeleteMovieWindowProps> = (args) => (
  <DeleteMovieWindow {...args} />
);

export const HideWindow = Template.bind({});
HideWindow.args = {
  showDeleteMovieModal: false,
};

export const ShowWindow = Template.bind({});
ShowWindow.args = {
  showDeleteMovieModal: true,
};
