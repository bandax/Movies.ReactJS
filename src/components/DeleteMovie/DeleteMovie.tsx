import * as React from 'react';
import './DeleteMovie.scss';
import { IMovieData } from '../../interfaces/IMovieData';

interface IDeleteMovieWindowProps {
  showDeleteMovieModal: boolean;
  movieId: number;
  onShowDeleteMovieWindow: (
    e: React.MouseEvent<HTMLButtonElement> | React.MouseEvent<HTMLAnchorElement>
  ) => void;
  onConfirmDeleteMovie: (id: number) => void;
}

const DeleteMovieWindow: React.FunctionComponent<IDeleteMovieWindowProps> = (
  props: IDeleteMovieWindowProps
) => {
  const onSubmitClicked = (e: React.FormEvent<HTMLButtonElement>) => {
    e.preventDefault();
    props.onConfirmDeleteMovie(props.movieId);
  };

  if (!props.showDeleteMovieModal) {
    return null;
  }
  return (
    <div className="delete-movie-window">
      <div className="delete-movie-modal">
        <h2 className="delete-movie-title">Delete movie</h2>
        <a
          className="delete-movie-close"
          href="#"
          onClick={props.onShowDeleteMovieWindow}
        >
          &times;
        </a>
        <div className="delete-movie-content">
          Are you sure you want to delete this movie?
        </div>
        <div className="delete-movie-actions">
          <button className="btn btn-confirm" onClick={onSubmitClicked}>
            Confirm
          </button>
        </div>
      </div>
    </div>
  );
};

export { DeleteMovieWindow };
