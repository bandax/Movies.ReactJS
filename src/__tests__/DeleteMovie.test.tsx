import * as React from 'react';
import { DeleteMovieWindow } from '../components/DeleteMovie/DeleteMovie';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';

describe('<DeleteMovieWindow />', () => {
  let onShowDeleteMovieWindow;
  let onConfirmDeleteMovie;

  beforeEach(() => {
    onShowDeleteMovieWindow = jest.fn();
    onConfirmDeleteMovie = jest.fn();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('render a Delete Movie Component', async () => {
    const { asFragment } = render(
      <DeleteMovieWindow
        showDeleteMovieModal={true}
        movieId={1234}
        onShowDeleteMovieWindow={null}
        onConfirmDeleteMovie={null}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not render the component at all', async () => {
    const { container } = render(
      <DeleteMovieWindow
        showDeleteMovieModal={false}
        movieId={1234}
        onShowDeleteMovieWindow={null}
        onConfirmDeleteMovie={null}
      />
    );

    const elements = container.firstChild;

    expect(elements).toBeNull();
  });

  it('should close add movie window when click close icon', async () => {
    const { getByTestId, getByText } = render(
      <DeleteMovieWindow
        showDeleteMovieModal={true}
        movieId={1234}
        onShowDeleteMovieWindow={onShowDeleteMovieWindow}
        onConfirmDeleteMovie={null}
      />
    );
    const button = getByTestId('delete-movie-close');

    userEvent.click(button);

    expect(getByText('Delete movie')).toBeInTheDocument();
    await waitFor(() => {
      expect(onShowDeleteMovieWindow).toHaveBeenCalled();
    });
  });

  it('should call delete movie method when submit button is clicked', async () => {
    const movieId = 1234;
    const { getByTestId, getByText } = render(
      <DeleteMovieWindow
        showDeleteMovieModal={true}
        movieId={movieId}
        onShowDeleteMovieWindow={onShowDeleteMovieWindow}
        onConfirmDeleteMovie={onConfirmDeleteMovie}
      />
    );
    const button = getByText('Confirm');

    userEvent.click(button);

    expect(getByText('Delete movie')).toBeInTheDocument();
    await waitFor(() => {
      expect(onConfirmDeleteMovie).toHaveBeenCalledWith(movieId);
    });
  });
});
