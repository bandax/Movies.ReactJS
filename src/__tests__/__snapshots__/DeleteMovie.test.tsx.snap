// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteMovieWindow /> render a Delete Movie Component 1`] = `
<DocumentFragment>
  <div
    class="delete-movie-window"
  >
    <div
      class="delete-movie-modal"
    >
      <h2
        class="delete-movie-title"
      >
        Delete movie
      </h2>
      <a
        class="delete-movie-close"
        data-testid="delete-movie-close"
        href="#"
      >
        Ã—
      </a>
      <div
        class="delete-movie-content"
      >
        Are you sure you want to delete this movie?
      </div>
      <div
        class="delete-movie-actions"
      >
        <button
          class="btn btn-confirm"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
